src/GapicClientTrait.php

```diff
@@ -528,6 +528,98 @@ private static function defaultTransport()
             : 'rest';
     }
 
+    private function validateCallConfig(string $methodName)
```

Comment:
Outside of checking for the method name, what scenarios do we expect to need to run this validation? Since we're generating the descriptor configs ourselves and they aren't user input, I'm just thinking through if we need to do this. It seems like we'd have bigger problems if these fields don't exist.

Reply:
For me, it is more about providing a better experience for those error scenarios than it is about preventing them. I'm not sure if it is entirely possible, but if an older version of a client uses a newer version of gax, and a manual layer there attempts to make use of these newer gax APIs, I think this exception is better than a null pointer or segfault.

Reply:
The vector seems unlikely, but the cost of doing these checks is cheap so if you prefer them I'm okay with it.